web/main/models.py

```diff
@@ -2403,8 +2444,22 @@ def _delete_related_links_and_text_blocks(self):
     def get_slug(self):
         return slugify(self.title)
 
-    def viewable_by(self, user):
-        return self.casebook.viewable_by(user)
+    def viewable_by(self, user: User) -> bool:
+        """A node may belong to a viewable casebook, but itself not be viewable because of professor-only content"""
+
+        if not self.casebook:  # An orphaned node isn't viewable, presumably
```

Comment:
This is mostly a sop to the type checker. A `ContentNode` does not actually need to belong to a `Casebook` according to the data model (it's `null=True`), though in practice I'm not sure if that happens. Maybe it's for cloning by node?

Reply:
A `ContentNode` that does not belong to a casebook is a relic of times past, when there was no `Casebook` model: those nodes were the top-level of a a casebook, the root, a Casebook-flavored `ContentNode`, as opposed to a Section- or Resource-flavored node. It's what the [`old_casebook`](https://github.com/harvard-lil/h2o/blob/develop/web/main/models.py#L3474) field on the `Casebook` model points too. We checked recently; as of that check, that accounted for all `ContentNode` objects that didn't belong to casebooks except one. That singleton was clearly a messed up thing made by a programmer in the shell, and not an artifact of any real book.

Reply:
This is a thing I'm hoping to clean up soon.